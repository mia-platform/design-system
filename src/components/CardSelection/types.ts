
export enum Type {
  Radio = 'radio',
  Checkbox = 'checkbox'
}
