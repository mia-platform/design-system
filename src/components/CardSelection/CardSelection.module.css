
.card {
  position: relative;

  border-radius: var(--shape-border-radius-lg, 8px);
  background-color: var(--palette-action-alternate-secondary-default, #89898914);
  color: var(--palette-text-neutral-main, #636363);
  transition: all 0.1s cubic-bezier(0.71, -0.46, 0.88, 0.6);

  padding: var(--spacing-padding-lg, 8px);
  gap: var(--spacing-gap-sm, 8px);

  display: flex;
  flex-direction: column;
  align-items: center;

  cursor: pointer;

  &:hover:not(.disabled) {
    color: var(--palette-text-neutral-bold, #464646);
  }
}

.selected:not(.disabled) {
  background-color: var(--palette-action-alternate-primary-selected, #1890FF29);
  color: var(--palette-text-primary-bold, #004AAA) !important;

  &:hover {
    color: var(--palette-text-primary-subtle, #1890FF) !important;
    & :global(.mia-platform-checkbox) :global(.mia-platform-checkbox-inner) {
      border-color: var(--palette-action-primary-hover, #1890FF) !important;
      background-color: var(--palette-action-primary-hover, #1890FF) !important;
    }
    & :global(.mia-platform-radio-inner) {
      border-color: var(--palette-action-primary-hover, #1890FF) !important;
    }
    & :global(.mia-platform-radio-inner:after) {
      background-color: var(--palette-action-primary-hover, #1890FF) !important;
    }
  }
}

.disabled {
  cursor: not-allowed;
  color: var(--palette-text-neutral-subtler, #ACACAC);

  & .subtitle {
    color: var(--palette-text-neutral-subtler, #ACACAC);
  }
}


.horizontal {
  align-items: flex-start;
  .header {
    flex-direction: row;
  }
}

.header {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  gap: var(--spacing-gap-sm, 8px);
}

.title {
  font-size: var(--typography-bodyMBold-fontSize, 16px);
  line-height: var(--typography-bodyMBold-lineHeight, 24px);
  font-weight: var(--typography-bodyMBold-fontWeight, 600);
}

.subtitle {
  font-size: var(--typography-bodyS-fontSize, 14px);
  line-height: var(--typography-bodyS-lineHeight, 16px);
  font-weight: var(--typography-bodyS-fontWeight, 400);
  color: var(--palette-text-neutral-main, #636363);

  text-align: center;
}

.actionContainer {
  display: flex;
  justify-content: center;
  position: absolute;
  top: 20px;
  right: 20px;
}

.content {
  margin: var(--spacing-margin-sm, 8px);
}
