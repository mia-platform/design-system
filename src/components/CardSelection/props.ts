/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactNode } from 'react'

import { InputType, Layout } from './types.ts'
import { IconComponent } from '../Icon/Icon.props.ts'

export type CardSelectionOption<T> = {

  /**
   * Optional id value for the card.
   */
  id?: string

  /**
   * The main title displayed prominently on the card.
   */
  title: ReactNode

  /**
   * An optional icon displayed alongside the title.
   */
  icon?: IconComponent

  /**
   * A subtitle displayed below the title.
   */
  subtitle?: ReactNode

  /**
   * Custom content to be rendered inside the card.
   */
  content?: ReactNode

  /**
   * The value associated with the card.
   */
  value?: T

  /**
   * Determines if the card is disabled.
   */
  disabled?: boolean
}

export type CardSelectionProps<T> = {

  /**
   * The default value for the card selection.
   * Must be an array if inputType is "checkbox", a value if inputType is "radio".
   */
  defaultValue?: T | T[]

  gap?: number

  /**
   * The inputType of the card selection. Defaults: undefined (no input will be displayed).
   */
  inputType?: InputType,

  isDisabled?: boolean

  /**
   * The layout of the cards. Can be "horizontal" or "vertical". Default: "vertical".
   */
  layout: Layout

  /*
  * Handler triggered when the value of the selection changes (only triggered if inputType is "checkbox" or "radio")
  * */
  onChange?: (value: T | T[]) => void

  /*
  * Handler triggered when a card is clicked
  * */
  onClick?: (value?: T) => void

  /**
   * The options array for the card selection.
   */
  options: CardSelectionOption<T>[]

  /**
   * The controlled value for the card selection.
   * Must be an array if inputType is "checkbox", a value if inputType is "radio".
   */
  value?: T | T[]
}
