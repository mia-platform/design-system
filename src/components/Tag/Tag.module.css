.tag {
  margin-inline-end: 0px;
}
