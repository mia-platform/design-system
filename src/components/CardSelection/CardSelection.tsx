/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, useCallback, useMemo, useRef, useState } from 'react'
import classnames from 'classnames'

import { CardSelectionProps } from './props.ts'
import { Checkbox } from '../Checkbox'
import { Icon } from '../Icon'
import { Radio } from '../RadioGroup/components/Radio'
import { Type } from './types.ts'
import styles from './CardSelection.module.css'

export const defaults = {}

export const CardSelection = <T, >({
  children,
  horizontal,
  type,
  icon,
  title,
  subtitle,
  value,
  isDisabled,
  isChecked: isCheckedProp,
  isInitiallyChecked,
  onClick,
}: CardSelectionProps<T>): ReactElement => {
  const [isCheckedState, setIsCheckedState] = useState(isInitiallyChecked)
  const isControlled = useRef(isCheckedProp !== undefined).current

  const isChecked = isControlled ? isCheckedProp : isCheckedState

  const className = useMemo(() => classnames([
    styles.card,
    horizontal && styles.horizontal,
    isChecked && styles.selected,
    isDisabled && styles.disabled,
  ]), [horizontal, isChecked, isDisabled])

  const action = useMemo(() => {
    switch (type) {
    case Type.Radio:
      return <Radio isChecked={isChecked} isDisabled={isDisabled} />
    case Type.Checkbox:
      return <Checkbox isChecked={isChecked} isDisabled={isDisabled} />
    default:
      return undefined
    }
  }, [isChecked, isDisabled, type])

  const handleClick = useCallback(() => {
    const nextChecked = !isChecked
    if (!isControlled) {
      setIsCheckedState(nextChecked)
    }
    if (onClick) {
      onClick(nextChecked, value)
    }
  }, [isChecked, isControlled, onClick, value])

  return (
    <div className={className} onClick={!isDisabled ? handleClick : undefined}>
      <div className={styles.header}>
        {
          icon && <Icon component={icon} size={horizontal ? 24 : 32} />
        }
        <span className={styles.title}>{title}</span>
      </div>
      {subtitle && <span className={styles.subtitle}>{subtitle}</span>}
      {
        action && (
          <div className={styles.actionContainer}>
            {action}
          </div>
        )
      }
      {children}
    </div>
  )
}

CardSelection.Type = Type
