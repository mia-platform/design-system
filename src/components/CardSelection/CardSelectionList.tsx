import { Checkbox, Radio, RadioChangeEvent } from 'antd'
import { ComponentType, ReactElement, useCallback, useMemo } from 'react'

import { InputType, Layout } from './types.ts'
import { CardSelection } from './index.ts'
import { Props } from './props.ts'
import styles from './CardSelection.module.css'

const defaults = {
  gap: 24,
  layout: Layout.Vertical,
}

export const CardSelectionList = <T, >({
  type,
  layout = defaults.layout,
  gap = defaults.gap,
  options,
  isDisabled,
  onChange,
}: Props<T>): ReactElement => {
  const isInput = type === InputType.Radio || type === InputType.Checkbox

  const Component = useMemo(() => {
    switch (type) {
    case InputType.Checkbox:
      return Checkbox.Group
    case InputType.Radio:
      return Radio.Group
    default:
      return 'div' as unknown as ComponentType
    }
  }, [type])

  const handleChange = useCallback((value: T[] | RadioChangeEvent) : void => {
    if (onChange) {
      onChange(Array.isArray(value) ? value : [value.target.value])
    }
  }, [onChange])

  const children = useMemo(
    () => options.map(({ id, disabled, ...rest }, index) => (
      <CardSelection
        inputType={type}
        isDisabled={isDisabled || disabled}
        key={id || index}
        layout={layout}
        {...rest}
      />
    )), [isDisabled, layout, options, type])

  return (
    <Component
      className={styles.list}
      style={{ gap }}
      {...(isInput && {
        disabled: isDisabled,
        onChange: !isDisabled ? handleChange : undefined,
      })}
    >
      {children}
    </Component>
  )
}

CardSelectionList.Type = InputType
