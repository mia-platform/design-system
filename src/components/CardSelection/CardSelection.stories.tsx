/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { Meta, StoryObj } from '@storybook/react'
import { Flex } from 'antd'
import { PiPlaceholder } from 'react-icons/pi'

import { CardSelection, defaults } from './CardSelection'
import { Tag } from '../Tag'

const defaultProps = {
  title: 'Title',
  subtitle: 'Subtitle',
  icon: PiPlaceholder,
  children: (
    <Flex gap={8}>
      <Tag color="blue">Tag name</Tag>
      <Tag color="red">Tag name</Tag>
    </Flex>
  ),
}

const meta = {
  component: CardSelection,
  args: {
    ...defaults,
    ...defaultProps,
  },
} satisfies Meta<typeof CardSelection>

export default meta
type Story = StoryObj<typeof meta>

export const Vertical: Story = {
  args: {},
}

export const Horizontal: Story = {
  args: {
    horizontal: true,
  },
}

export const VerticalCheckbox: Story = {
  args: {
    type: CardSelection.Type.Checkbox,
  },
}

export const HorizontalCheckbox: Story = {
  args: {
    type: CardSelection.Type.Checkbox,
    horizontal: true,
  },
}

export const VerticalRadio: Story = {
  args: {
    type: CardSelection.Type.Radio,
  },
}

export const HorizontalRadio: Story = {
  args: {
    type: CardSelection.Type.Radio,
    horizontal: true,
  },
}

export const CheckboxDisabled: Story = {
  args: {
    type: CardSelection.Type.Checkbox,
    horizontal: true,
    isDisabled: true,
    isChecked: true,
  },
}

export const RadioDisabled: Story = {
  args: {
    type: CardSelection.Type.Radio,
    horizontal: true,
    isDisabled: true,
    isChecked: true,
  },
}
