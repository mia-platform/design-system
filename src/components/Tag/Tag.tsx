/**
 * Copyright 2024 Mia srl
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * SPDX-License-Identifier: Apache-2.0
 */

import { ReactElement, useMemo } from 'react'
import { Tag as AntTag } from 'antd'
import classnames from 'classnames'

import { Color, Type } from './types.ts'
import { Icon } from '../Icon'
import { TagProps } from './Tag.props'
import styles from './Tag.module.css'

export const defaults = {
  isBordered: true,
  color: Color.Grey,
}

export const Tag = (
  {
    children,
    type,
    color = Color.Grey,
    closeIcon,
    onClose,
    icon,
    isBordered = defaults.isBordered,
  } : TagProps
) : ReactElement => {
  const className = useMemo(() => classnames([
    styles.tag,
    type === Type.Chip && styles.chip,
    color === Color.Grey && styles.tagColorGrey,
    color === Color.Blue && styles.tagColorBlue,
    color === Color.Teal && styles.tagColorTeal,
    color === Color.Green && styles.tagColorGreen,
    color === Color.Yellow && styles.tagColorYellow,
    color === Color.Magenta && styles.tagColorMagenta,
    color === Color.Purple && styles.tagColorPurple,
    color === Color.Red && styles.tagColorRed,
  ]), [color, type])

  const customColor = useMemo(() => {
    return color && !(color in Color) ? color : undefined
  }, [color])

  return (
    <AntTag
      bordered={isBordered}
      className={className}
      closeIcon={closeIcon}
      color={customColor}
      icon={icon && (
        <Icon component={icon} size={16} />
      )}
      onClose={onClose}
    >
      {children}
    </AntTag>
  )
}

Tag.Type = Type
Tag.Color = Color
