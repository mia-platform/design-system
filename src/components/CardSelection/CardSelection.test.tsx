import { Flex } from 'antd'
import { PiPlaceholder } from 'react-icons/pi'
import { fireEvent } from '@testing-library/dom'

import { CardSelection, defaults } from './CardSelection'
import { render, screen } from '../../test-utils.tsx'
import { Tag } from '../Tag'

// Default props for the CardSelection component
const defaultProps = {
  ...defaults,
  title: 'Title',
  subtitle: 'Subtitle',
  icon: PiPlaceholder,
  children: (
    <Flex gap={8}>
      <Tag color="blue">Tag name</Tag>
      <Tag color="red">Tag name</Tag>
    </Flex>
  ),
}

describe('CardSelection Snapshot Tests', () => {
  it('renders Vertical correctly', () => {
    const { asFragment } = render(<CardSelection {...defaultProps} />)
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders Horizontal correctly', () => {
    const { asFragment } = render(
      <CardSelection {...defaultProps} horizontal={true} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders VerticalCheckbox correctly', () => {
    const { asFragment } = render(
      <CardSelection {...defaultProps} type={CardSelection.Type.Checkbox} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders HorizontalCheckbox correctly', () => {
    const { asFragment } = render(
      <CardSelection
        {...defaultProps}
        horizontal={true}
        type={CardSelection.Type.Checkbox}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders VerticalRadio correctly', () => {
    const { asFragment } = render(
      <CardSelection {...defaultProps} type={CardSelection.Type.Radio} />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders HorizontalRadio correctly', () => {
    const { asFragment } = render(
      <CardSelection
        {...defaultProps}
        horizontal={true}
        type={CardSelection.Type.Radio}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders CheckboxDisabled correctly', () => {
    const { asFragment } = render(
      <CardSelection
        {...defaultProps}
        horizontal={true}
        isChecked={true}
        isDisabled={true}
        type={CardSelection.Type.Checkbox}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('renders RadioDisabled correctly', () => {
    const { asFragment } = render(
      <CardSelection
        {...defaultProps}
        horizontal={true}
        isChecked={true}
        isDisabled={true}
        type={CardSelection.Type.Radio}
      />
    )
    expect(asFragment()).toMatchSnapshot()
  })

  test('should call on click and change input value', () => {
    const onClick = jest.fn()
    const value = jest.fn()
    render(
      <CardSelection
        {...defaultProps}
        type={CardSelection.Type.Radio}
        value={value}
        onClick={onClick}
      />
    )
    fireEvent.click(screen.getByRole('img'))
    expect(onClick).toHaveBeenCalledWith(true, value)
    expect(screen.getByRole('radio')).toBeChecked()
  })

  test('should not call on click and change input value if disabled', () => {
    const onClick = jest.fn()
    render(
      <CardSelection
        {...defaultProps}
        isDisabled
        type={CardSelection.Type.Radio}
        onClick={onClick}
      />
    )
    fireEvent.click(screen.getByRole('img'))
    expect(onClick).not.toHaveBeenCalled()
    expect(screen.getByRole('radio')).not.toBeChecked()
  })

  test('should accept a controlled value and render correctly', () => {
    const onClick = jest.fn()
    render(
      <CardSelection
        {...defaultProps}
        isChecked
        isDisabled
        type={CardSelection.Type.Radio}
        onClick={onClick}
      />
    )

    expect(screen.getByRole('radio')).toBeChecked()
  })
})
