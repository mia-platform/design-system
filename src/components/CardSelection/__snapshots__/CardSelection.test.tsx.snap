// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardSelection Snapshot Tests renders CheckboxDisabled correctly 1`] = `
<DocumentFragment>
  <div
    class="card horizontal selected disabled"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="actionContainer"
    >
      <div
        class="checkboxComponent disabled"
      >
        <div
          class="checkboxTextWrapper"
        >
          <label
            class="mia-platform-checkbox-wrapper mia-platform-checkbox-wrapper-checked mia-platform-checkbox-wrapper-disabled"
          >
            <span
              class="mia-platform-checkbox ant-wave-target mia-platform-checkbox-checked mia-platform-checkbox-disabled"
            >
              <input
                checked=""
                class="mia-platform-checkbox-input"
                disabled=""
                type="checkbox"
              />
              <span
                class="mia-platform-checkbox-inner"
              />
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders Horizontal correctly 1`] = `
<DocumentFragment>
  <div
    class="card horizontal"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders HorizontalCheckbox correctly 1`] = `
<DocumentFragment>
  <div
    class="card horizontal"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="actionContainer"
    >
      <div
        class="checkboxComponent"
      >
        <div
          class="checkboxTextWrapper"
        >
          <label
            class="mia-platform-checkbox-wrapper"
          >
            <span
              class="mia-platform-checkbox ant-wave-target"
            >
              <input
                class="mia-platform-checkbox-input"
                type="checkbox"
              />
              <span
                class="mia-platform-checkbox-inner"
              />
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders HorizontalRadio correctly 1`] = `
<DocumentFragment>
  <div
    class="card horizontal"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="actionContainer"
    >
      <div
        class="radio"
      >
        <label
          class="mia-platform-radio-wrapper"
        >
          <span
            class="mia-platform-radio ant-wave-target"
          >
            <input
              class="mia-platform-radio-input"
              type="radio"
              value=""
            />
            <span
              class="mia-platform-radio-inner"
            />
          </span>
        </label>
      </div>
    </div>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders RadioDisabled correctly 1`] = `
<DocumentFragment>
  <div
    class="card horizontal selected disabled"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="24"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="actionContainer"
    >
      <div
        class="radio"
      >
        <label
          class="mia-platform-radio-wrapper mia-platform-radio-wrapper-checked mia-platform-radio-wrapper-disabled"
        >
          <span
            class="mia-platform-radio ant-wave-target mia-platform-radio-checked mia-platform-radio-disabled"
          >
            <input
              checked=""
              class="mia-platform-radio-input"
              disabled=""
              type="radio"
              value=""
            />
            <span
              class="mia-platform-radio-inner"
            />
          </span>
        </label>
      </div>
    </div>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders Vertical correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="32"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders VerticalCheckbox correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="32"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="actionContainer"
    >
      <div
        class="checkboxComponent"
      >
        <div
          class="checkboxTextWrapper"
        >
          <label
            class="mia-platform-checkbox-wrapper"
          >
            <span
              class="mia-platform-checkbox ant-wave-target"
            >
              <input
                class="mia-platform-checkbox-input"
                type="checkbox"
              />
              <span
                class="mia-platform-checkbox-inner"
              />
            </span>
          </label>
        </div>
      </div>
    </div>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;

exports[`CardSelection Snapshot Tests renders VerticalRadio correctly 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="header"
    >
      <svg
        color="currentColor"
        fill="currentColor"
        height="32"
        role="img"
        stroke="currentColor"
        stroke-width="0"
        style="color: currentColor;"
        viewBox="0 0 256 256"
        width="32"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M208,32H48A16,16,0,0,0,32,48V208a16,16,0,0,0,16,16H208a16,16,0,0,0,16-16V48A16,16,0,0,0,208,32ZM48,208V59.31L196.69,208ZM59.31,48H208V196.7Z"
        />
      </svg>
      <span
        class="title"
      >
        Title
      </span>
    </div>
    <span
      class="subtitle"
    >
      Subtitle
    </span>
    <div
      class="actionContainer"
    >
      <div
        class="radio"
      >
        <label
          class="mia-platform-radio-wrapper"
        >
          <span
            class="mia-platform-radio ant-wave-target"
          >
            <input
              class="mia-platform-radio-input"
              type="radio"
              value=""
            />
            <span
              class="mia-platform-radio-inner"
            />
          </span>
        </label>
      </div>
    </div>
    <div
      class="mia-platform-flex"
      style="gap: 8px;"
    >
      <span
        class="mia-platform-tag mia-platform-tag-blue tag"
      >
        Tag name
      </span>
      <span
        class="mia-platform-tag mia-platform-tag-red tag"
      >
        Tag name
      </span>
    </div>
  </div>
</DocumentFragment>
`;
